environment_variable.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2016 Rohith All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package marathon
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// EnvironmentVariable is an environment variable associated with an application
+type EnvironmentVariable struct {
```

Comment:
The problem is that they aren't just overloading the environment variables, they're creating a new entry for Secrets in the app definition. See [here](https://docs.mesosphere.com/1.9/security/secrets/use-secrets/). An example config is: 

```json
{  
  "id":"/developer/service",
  "cmd":"sleep 100",
  "env":{  
     "MY_SECRET":{  
        "secret":"secret0"
     }
  },
  "secrets":{  
     "secret0":{  
        "source":"developer/my-secret"
     }
  }
}
```

The environment secret references a secret in the secrets section, which points to where the secret is stored in the secret store.  So neither the env part or the secrets part hold all the information for the secret.

I suppose the Secret could be a ```map{string}struct```.  Where the key is 'secret0' and the struct is:

```go
type struct{
    EnvVar string `json=",omitempty"`,
    Source string `json="source"`,
}
```
Then the custom unmarshaler for Env could store a secrets environment variable in the above struct, and a custom unmarshaler for Secret could assign the Source.  Then the marshaler for Env would need to be something like another struct to put the environment variable part of the secret in.

It seems a bit hacky to me, but let me know what you think.  I have some time to try to get it to work.